src/Maps/components/MapsContainer.ts

```diff
@@ -0,0 +1,242 @@
+        const commonProps = {
+            allLocations: this.state.locations,
+            fetchingData: this.state.isFetchingData,
+            className: this.props.class,
+            alertMessage: this.state.alertMessage,
+            divStyles: parseStyle(this.props.style),
+            onClickMarker: this.onClickMarker,
+            mapsToken: mapsApiToken,
+            ...this.props as MapProps
+        };
+
+        return this.props.mapProvider === "googleMaps"
+            ? createElement(googleApiWrapper, { ...commonProps })
+            : createElement(LeafletMap, { ...commonProps });
+    }
+
+    componentWillReceiveProps(nextProps: MapsContainerProps) {
+        this.resetSubscriptions(nextProps.mxObject);
+        const validationMessage = validateLocationProps(nextProps);
+        if (nextProps && nextProps.mxObject) {
+            if (validationMessage) {
+                this.setState({ alertMessage: validationMessage });
+            } else {
+                this.fetchData(nextProps.mxObject);
+            }
+        } else {
+            this.setState({ locations: [], alertMessage: "", isFetchingData: false });
+        }
+    }
+
+    componentWillUnmount() {
+        this.unsubscribeAll();
+    }
+
+    private resetSubscriptions(contextObject?: mendix.lib.MxObject) {
+        this.unsubscribeAll();
+        if (this.props.locations && this.props.locations.length) {
+            if (contextObject) {
+                this.subscriptionHandles.push(window.mx.data.subscribe({
+                    guid: contextObject.getGuid(),
+                    callback: () => this.fetchData(contextObject)
+                }));
+                this.props.locations.forEach(location => {
+                    this.subscriptionHandles.push(window.mx.data.subscribe({
+                        entity: location.locationsEntity as string,
+                        callback: () => this.fetchData(contextObject)
+                    }));
+                    [
+                        location.latitudeAttribute,
+                        location.longitudeAttribute,
+                        location.staticMarkerIcon,
+                        location.markerImageAttribute
+                    ]
+                    .forEach(
+                        (attr): number => this.subscriptionHandles.push(window.mx.data.subscribe({
+                            attr,
+                            callback: () => this.fetchData(contextObject),
+                            guid: contextObject.getGuid()
+                        }))
+                    );
+                });
+            }
+        }
+    }
+
+    private unsubscribeAll() {
+        this.subscriptionHandles.forEach(window.mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+    }
+
+    private fetchData = (contextObject?: mendix.lib.MxObject) => {
+        this.setState({ isFetchingData: true });
```

Comment:
Shouldn't the contextObject check be at this point?  Otherwise I find you're going to keep moving it lower and lower into other functions. I'd rather u do it at the very top, and handle it there as well. Here each iteration is likely going to be handling it.
`if (contenxtObject) {}`

Reply:
done